ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java

```diff
@@ -140,6 +144,12 @@ private static ContingencyResult buildContingencyResult(ContingencyResult optima
         return raer;
     }
 
+    private static List<RemedialActionElementResult> getRemedialActionElementResult(RangeAction rangeAction, Leaf optimalLeaf) {
+        String rangeActionId = rangeAction.getId();
+        RemedialActionResult remedialActionResult = optimalLeaf.getRaoResult().getPreContingencyResult().getRemedialActionResultById(rangeActionId);
```

Comment:
It will throw a FaraoException if the rangeAction is in the crac, but was not selected in the optimal leaf